auth-agents-common/src/main/java/org/apache/atlas/plugin/util/PolicyRefresher.java

```diff
@@ -147,16 +157,42 @@ public void setLastActivationTimeInMillis(long lastActivationTimeInMillis) {
 		this.lastActivationTimeInMillis = lastActivationTimeInMillis;
 	}
 
+	public void submitRefresherTask(boolean policies, boolean roles, boolean groups) {
+		if (adHocTaskLimiter.size() == 1) {
+			LOG.info("submitRefresherTask: Default refresh job is already submitted, skipping submitting another.");
+			return;
+		}
+
+		DownloaderTask task;
+
+		if (IS_RUNNING) {
+			LOG.info("submitRefresherTask: Another refresh task in already in progress, submitting another default task");
+			task = new DownloaderTask(policyDownloadQueue, true, true, true);
+			adHocTaskLimiter.add(task);
+		} else {
+			LOG.info("submitRefresherTask: Submitting custom task to refresh policies:{}, roles:{}, groups:{}",
+					policies, roles, groups);
+			task = new DownloaderTask(policyDownloadQueue, policies, roles, groups);
+		}
+
+		executor.submit(task);
+	}
+
 	public void startRefresher() {
-		loadRoles();
-		loadPolicy();
-		loadUserStore();
-		super.start();
+		try {
+			IS_RUNNING = true;
+			loadRoles();
+			loadPolicy();
+			loadUserStore();
+		} finally {
+			IS_RUNNING = false;
+		}
 
-		policyDownloadTimer = new Timer("policyDownloadTimer", true);
+		super.start();
 
 		try {
-			policyDownloadTimer.schedule(new DownloaderTask(policyDownloadQueue), pollingIntervalMs, pollingIntervalMs);
+			executor.scheduleWithFixedDelay(new DownloaderTask(policyDownloadQueue),
```

Comment:
this is fine, can you check if we can make use of `@Scheduled` which spring has. Else we are good with this. Just make sure you shutdown the pool

Reply:
Kept the same, just added code to shut down executorService